keps/sig-node/4639-oci-volume-source/README.md

```diff
@@ -0,0 +1,1009 @@
+checklist items _must_ be updated for the enhancement to be released.
+-->
+
+Items marked with (R) are required *prior to targeting to a milestone / release*.
+
+- [ ] (R) Enhancement issue in release milestone, which links to KEP dir in [kubernetes/enhancements] (not the initial KEP PR)
+- [ ] (R) KEP approvers have approved the KEP status as `implementable`
+- [ ] (R) Design details are appropriately documented
+- [ ] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input (including test refactors)
+  - [ ] e2e Tests for all Beta API Operations (endpoints)
+  - [ ] (R) Ensure GA e2e tests meet requirements for [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+  - [ ] (R) Minimum Two Week Window for GA e2e tests to prove flake free
+- [ ] (R) Graduation criteria is in place
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+The proposed enhancement adds a new `VolumeSource` to Kubernetes that supports OCI images and/or OCI artifacts.
+This allows users to package files and share them among containers in a pod without including them in the main image,
+thereby reducing vulnerabilities and simplifying image creation.
+
+While OCI images are well-supported by Kubernetes and CRI, extending support to
+OCI artifacts involves recognizing additional media types within container
+runtimes, implementing custom lifecycle management, resolution of artifact
+registry referrers use pattern for artifacts, and ensuring appropriate
+validation and security measures.
+
+## Motivation
+
+Supporting OCI images and artifacts directly as a `VolumeSource` allows
+Kubernetes to focus on OCI standards as well as allows to store and distribute
+any content using OCI registries. This allows the project to grow into use cases
+which go beyond running particular images.
+
+### Goals
+
+- Introduce a new `VolumeSource` type that allows mounting OCI images and/or artifacts.
```

Comment:
Aren't we saying though that we're going to (in this kep)

1. Mount OCI images
2. Delegate any unknown OCI media types to the runtime (thus shifting the burden to the container runtime?)

There seems to be some risks to 2 - how would we define conformance?  I would definitely expect an intree volume mount to clearly define conformance behavior for 1, but I can't quite describe it for 2.  However, the counter argument is that we don't define any of this for container images today (we simply conformance test on known OCI image types), and the exact same behavior would happen - as long as a filesystem exists after the OCI pull action is resolved, kube doesn't care.

So that to me feels like the bar - "a VolumeSource type that allows mounting OCI images and any future OCI artifacts that can be expanded to a file or directory representation"

Reply:
It may be easiest to just put other types of oci artifacts out of scope for now. When other oci artifact types that may make sense to mount as a filesystem show up, we consider them in a future kep?

Reply:
We could specify how the directory structure will look like. There are some considerations for how volumes will be shared with SELinux labels and we may need to overlay layer them but as long as final directory structure is the same across runtimes, that should work.

Reply:
I added a point to:

> Providing a runtime guideline of how artifact files and directories should be mounted.